data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -34,14 +43,98 @@
+        initializeIndexPrefixAndSuffix();
+    }
+
+    private void initializeIndexPrefixAndSuffix(){
+        String indexAliasFromConfig = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+        indexPrefix = indexAliasFromConfig.replaceAll(TIME_PATTERN_REGULAR_EXPRESSION, "");
+
+        final Pattern pattern = Pattern.compile(TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION);
+        final Matcher matcher = pattern.matcher(indexAliasFromConfig);
+        if (matcher.find()) {
+            String timePattern = matcher.group(1);
+            validateTimePatternIsAtTheEnd(indexAliasFromConfig, timePattern);
+            validateNoSpecialCharsInTimePattern(timePattern);
+            validateTimePatternNotTooGranular(timePattern);
+            indexTimeSuffixFormatter = Optional.of(DateTimeFormatter.ofPattern(timePattern));
+        }else{
+            indexTimeSuffixFormatter = Optional.empty();
+        }
+    }
+
+    /*
+      Data Prepper only allows time pattern as a suffix.
+     */
+    private void validateTimePatternIsAtTheEnd(final String indexAliasFromConfig, final String timePattern) {
+        if (!indexAliasFromConfig.endsWith(timePattern + "}")) {
+            throw new IllegalArgumentException("Time pattern can only be a suffix of a index.");
+        }
+    }
+
+    /*
+     * Special characters can cause failures in creating indexes.
+     * */
+    private static final Set<Character> INVALID_CHARS = ImmutableSet.of('#', '\\', '/', '*', '?', '"', '<', '>', '|', ',', ':');
+
+    private void validateNoSpecialCharsInTimePattern(final String timePattern) {
+        final boolean containsInvalidCharacter = timePattern.chars()
+                .mapToObj(c -> (char) c)
+                .anyMatch(character -> INVALID_CHARS.contains(character));
+        if (containsInvalidCharacter) {
+            throw new IllegalArgumentException("Index time pattern contains one or multiple special characters: " + INVALID_CHARS);
+        }
+    }
+
+    /*
+     * Data Prepper doesn't support creating indexes with time patterns that are too granular, e.g. minute, second, millisecond, nanosecond.
+     * https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+     * */
+    private static final Set<Character> UNSUPPORTED_TIME_GRANULARITY_CHARS = ImmutableSet.of('m', 's', 'S', 'A', 'n', 'N');
+
+    private void validateTimePatternNotTooGranular(final String timePattern) {
+        final boolean containsUnsupportedTimeSymbol = timePattern.chars()
+                .mapToObj(c -> (char) c)
+                .anyMatch(character -> UNSUPPORTED_TIME_GRANULARITY_CHARS.contains(character));
+        if (containsUnsupportedTimeSymbol) {
+            throw new IllegalArgumentException("To avoid creating a new index in less than one hour, " +
+                    "index time pattern should not contain a time symbol that are too granular: "
```

Comment:
I would keep the error messaging consistent and short like the other validations:
```
"Index time pattern contains time patterns that are less than one hour: + UNSUPPORTED_TIME_GRANULARITY_CHARS"
```

Reply:
Sounds good.